### DISCUSSION TITLE
Listening to part of state with subscribeWithSelector but access full state

### DISCUSSION THREAD

**Discussion by Robstei** [pmndrs/zustand#2064] [Q&A]
I want update part of the state when another part changes.

 That caused an infinity loop when using subscribe without subscribeWithSelector.

How does one access the full state while only listening to changes on "paw"?

Or should I use a different pattern?

From the docs:

``` javascript
import { subscribeWithSelector } from 'zustand/middleware'
const useDogStore = create(
  subscribeWithSelector(() => ({ paw: true, snout: true, fur: true }))
)

// Listening to selected changes, in this case when "paw" changes
const unsub2 = useDogStore.subscribe((state) => state.paw, console.log) <--- here
```

To make the example clearer: How would one flip the value of fur every time paw changes?

**Comment by dai-shi** [comment]
I would recommend to avoid using `subscribeWithSelector` and do things on your own. ex:
```js
const useDogStore = create(
  () => ({ paw: true, snout: true, fur: true })
)

// Listening to selected changes, in this case when "paw" changes
let lastPaw;
const unsub2 = useDogStore.subscribe((state) => {
  if (state.paw !== lastPaw) {
    lastPaw = state.paw;
    console.log(state);
  }
});
```

Or, you could always use `useDogStore.getState()` to get the full state.
